packages/@aws-cdk/aws-lambda-nodejs/test/bundling.test.ts

```diff
@@ -184,13 +187,17 @@ test('esbuild bundling with esbuild options', () => {
           'npx esbuild --bundle "/asset-input/lib/handler.ts"',
           '--target=es2020 --platform=node --outfile="/asset-output/index.js"',
           '--minify --sourcemap --external:aws-sdk --loader:.png=dataurl',
-          '--define:DEBUG=true --define:process.env.KEY="VALUE"',
+          defineInstructions,
           '--log-level=silent --keep-names --tsconfig=/asset-input/lib/custom-tsconfig.ts',
           '--metafile=/asset-output/index.meta.json --banner=\'/* comments */\' --footer=\'/* comments */\'',
         ].join(' '),
       ],
     }),
   });
+
+  // Make sure that the define instructions are working as expected with the esbuild CLI
```

Comment:
Do you think we really need to run `esbuild` here in the test? Isn't it enough to ensure that the right command line args are passed?

Reply:
Well, I got it right this time and reverse-engineered the define parameters' expectation based on that (it's a hell of escape slashes). It should be stable now and the test is not exactly necessary, but it's proof that it works now.

Let me know if you want me to remove it 👍🏻

Reply:
I like that this is doing an end to end test